app/models/supportconfig.rb

```diff
@@ -0,0 +1,46 @@
+require 'cheetah'
+
+class Supportconfig
+  class NotGeneratedError < StandardError; end
+
+  attr_reader :generated_at
+
+  def initialize(dir: Rails.root.join('public'))
+    @dir = dir
+    @generated_at = KeyValue.get('supportconfig_generated_at')
+  end
+
+  def generate()
+    stdout, stderr = Cheetah.run(
+      %W(
+        supportconfig -Q -g -k
+        -B #{self.filename_element}
+        -R #{@dir}
+      ),
+      stdout: :capture,
+      stderr: :capture,
+      logger: Logger.new(Rails.configuration.cli_log)
+    )
+    @generated_at = DateTime.now
```

Comment:
could be a bool instead of recording the time, unless the plan is to show the generation time in the UI, which is fine.